### PR TITLE
DEVPROD-835: Add "Depends On" tooltip

### PR DESCRIPTION
**Pull request by sophstad**
DEVPROD-835

### Description

For tasks in the task table that are blocked, show blocking tasks in a tooltip when the badge is hovered over

### Screenshots

image

### Testing

- Add Cypress test

### CODE REVIEW

File: src/components/TasksTable/index.tsx [hunk 1]
```diff
@@ -172,9 +175,24 @@ const getColumnDefs = ({
       multiple: 4,
     },
     className: "cy-task-table-col-STATUS",
-    render: (status: string, { execution, id }) =>
-      status && (
-        <TaskStatusBadge status={status} id={id} execution={execution} />
+    render: (status: string, { dependsOn, execution, id }) =>
+      dependsOn?.length && status === TaskStatus.Blocked ? (
+        <Tooltip
```
Comment (minnakt) [inline src/components/TasksTable/index.tsx:180]: You may have to tweak the z-index (the table icons overlap it)!:

Screenshot 2024-02-14 at 10 27 23 AM

File: src/components/TasksTable/index.tsx [hunk 2]
```diff
@@ -172,9 +175,24 @@ const getColumnDefs = ({
...
+          data-cy="depends-on-tooltip"
+          justify="middle"
+          trigger={
+            <span>
+              <TaskStatusBadge status={status} id={id} execution={execution} />
+            </span>
+          }
+        >
+          Depends on <>{pluralize("task", dependsOn.length)}</>:{" "}
```
Comment (minnakt) [inline src/components/TasksTable/index.tsx:189]: Is the fragment necessary?
```suggestion
          Depends on {pluralize("task", dependsOn.length)}:{" "}
```

**Review by minnakt — APPROVED** [review]

**Review by khelif96 — COMMENTED** [review]
image

How do you feel about updating the design a bit so that it shows the tasks that are actually being depended on and their statuses?

**Comment by sophstad** [comment]
> image
> How do you feel about updating the design a bit so that it shows the tasks that are actually being depended on and their statuses?

Are you referring to displaying `metStatus`? The actual status of the depended on task isn't queryable at the moment. ([schema](https://github.com/evergreen-ci/evergreen/blob/70172791d8c8b43b56506502ea75aaef7d204f08/graphql/schema/types/task.graphql#L135-L141))

**Comment by khelif96** [comment]
> > image
> > How do you feel about updating the design a bit so that it shows the tasks that are actually being depended on and their statuses?
> 
> Are you referring to displaying `metStatus`? The actual status of the depended on task isn't queryable at the moment. ([schema](https://github.com/evergreen-ci/evergreen/blob/70172791d8c8b43b56506502ea75aaef7d204f08/graphql/schema/types/task.graphql#L135-L141))

Yes MetStatus since that indicates which dependency has not met the expected status.

**Pull request merged**